xlive/H2MOD/Modules/MapManager/MapManager.cpp

```diff
@@ -255,55 +233,49 @@ char __cdecl handle_map_download_callback() {
 	return 1;
 }
 
-typedef char(__cdecl* leave_game_callback_def)();
-leave_game_callback_def p_original_leave_game_callback;
+typedef bool(__cdecl* map_download_leave_session_cb_t)();
+map_download_leave_session_cb_t p_map_download_leave_session_cb;
 
-char leavegame_callback()
+bool map_download_dialog_callback()
 {
-	return p_original_leave_game_callback();
+	return p_map_download_leave_session_cb();
 }
 
 /**
 * Menu constructor hook
 */
-void __cdecl display_map_downloading_menu(int a1, int a2, int a3, __int16 a4, DWORD map_download_callback, DWORD leave_game_callback, int a7, int a8, int a9, int a10)
+void __cdecl ui_open_map_download_confirm_dialog(int32 a1, int32 a2, int32 a3, int16 a4, void* map_download_callback, void* leave_game_callback, int32 a7, int32 a8, int32 a9, int32 a10)
 {
-	typedef void(__cdecl map_downloading_menu_constructor)(int a1, int a2, int a3, __int16 a4, int a5, int a6, int a7, int a8, int a9, int a10);
-	auto p_map_downloading_menu_constructor = Memory::GetAddress<map_downloading_menu_constructor*>(0x20E2E0);
-
-	p_original_leave_game_callback = (leave_game_callback_def)leave_game_callback;
-	p_map_downloading_menu_constructor(a1, a2, a3, a4, (DWORD)handle_map_download_callback, (DWORD)leavegame_callback, a7, a8, a9, a10);
+	p_map_download_leave_session_cb = (map_download_leave_session_cb_t)leave_game_callback;
+	INVOKE(0x20E2E0, 0x0, ui_open_map_download_confirm_dialog, a1, a2, a3, a4, handle_map_download_callback, map_download_dialog_callback, a7, a8, a9, a10);
 }
 
-int __cdecl get_total_map_downloading_percentage()
+int32 __cdecl get_total_map_downloading_percentage()
 {
 	if (!mapManager->m_mapDownloadQueryList.empty())
 		return mapManager->m_mapDownloadQueryList.front()->GetDownloadPercentage();
 	else
 		return 0;
 }
 
-const wchar_t* receiving_map_string[] = {
-	L"You are receiving the map from %s. \r\nPlease wait...%i%%"
-};
-
 const wchar_t* get_receiving_map_string()
 { 
-	int(__cdecl* get_default_game_language)() = (int(__cdecl*)())((char*)Memory::GetAddress(0x381fd));
-	wchar_t** receiving_map_message = Memory::GetAddress<wchar_t**>(0x46575C);
+	e_language current_language = get_current_language();
+	wchar_t** map_receiving_default_text = Memory::GetAddress<wchar_t**>(0x46575C);
 
-	if (get_default_game_language() == 0) // check if english
-		return receiving_map_string[0];
+	if (current_language == _language_english) // check if english
```

Comment:
This doesn't need to be here since we're getting the array from ingame regardless